plugins/modules/clustering/consul/consul.py

```diff
@@ -589,15 +573,27 @@ def main():
             token=dict(no_log=True)
         ),
         supports_check_mode=False,
+        mutually_exclusive=[
+            ('script', 'ttl', 'tcp', 'http'),
```

Comment:
Right now these are only mutually exclusive when `state=present`. So technically this is a breaking change.

Reply:
As far as I could tell these parameters are **only used when** `state=present`.

Reply:
Yes, but users can still specify them when `state=absent`. If you specified all of them for `state=absent` the module just worked fine so far. With your change, it will suddenly be erroring out.

Reply:
I sometimes think the lengths we go to keep *some* of the backwards compatibility to be a bit too far. But that's just me.

Okie, so let's take the long road: will pop a deprecation warning if any of those parameters are used with `state=absent`.